packages/react/src/PageHeader/PageHeader.tsx

```diff
@@ -651,39 +741,107 @@ const Navigation: React.FC<React.PropsWithChildren<NavigationProps>> = ({
   'aria-label': ariaLabel,
   'aria-labelledby': ariaLabelledBy,
 }) => {
+  const enabled = useFeatureFlag(CSS_MODULES_FEATURE_FLAG)
   warning(
     as === 'nav' && !ariaLabel && !ariaLabelledBy,
     'Use `aria-label` or `aria-labelledby` prop to provide an accessible label to the `nav` landmark for assistive technology',
   )
-
   return (
     <Box
       as={as}
       // Render `aria-label` and `aria-labelledby` only on `nav` elements
       aria-label={as === 'nav' ? ariaLabel : undefined}
       aria-labelledby={as === 'nav' ? ariaLabelledBy : undefined}
-      className={className}
-      sx={merge<BetterSystemStyleObject>(
-        {
-          gridRow: GRID_ROW_ORDER.Navigation,
-          gridArea: 'navigation',
-          paddingTop: '0.5rem',
-          display: 'flex',
-          ...getBreakpointDeclarations(hidden, 'display', value => {
-            return value ? 'none' : 'block'
-          }),
-          fontWeight: 'initial',
-          lineHeight: 'var(--text-body-lineHeight-medium, 1.4285)',
-          fontSize: 'var(--text-body-size-medium, 0.875rem)',
-        },
-        sx,
-      )}
+      className={clsx(enabled && classes.Navigation, className)}
+      sx={
+        enabled
+          ? sx
+          : merge<BetterSystemStyleObject>(
+              {
+                gridRow: GRID_ROW_ORDER.Navigation,
+                gridArea: 'navigation',
+                paddingTop: '0.5rem',
+                display: 'flex',
+                ...getBreakpointDeclarations(hidden, 'display', value => {
+                  return value ? 'none' : 'block'
+                }),
+                fontWeight: 'initial',
+                lineHeight: 'var(--text-body-lineHeight-medium, 1.4285)',
+                fontSize: 'var(--text-body-size-medium, 0.875rem)',
+              },
+              sx,
+            )
+      }
+      {...getHiddenDataAttributes(enabled, hidden)}
     >
       {children}
     </Box>
   )
 }
 
+// Based on getBreakpointDeclarations, this function will return the
+// correct data attribute for the given hidden value for CSS modules.
+function getHiddenDataAttributes(
+  isCssModules: boolean,
+  isHidden: boolean | ResponsiveValue<boolean>,
+): {
+  'data-is-hidden-all'?: boolean
```

Comment:
Such a tiny nitpick and I can't tell if this is new or existed already, but I would just name these `data-hidden-X` and remove "is" 😄